src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/SparkSessionTests.cs

```diff
@@ -65,5 +68,93 @@ public void TestSignaturesV2_4_X()
         }
+
+        /// <summary>
+        /// Test CreateDataFrame APIs.
+        /// </summary>        
+        [Fact]
+        public void TestCreateDataFrame()
+        {            
+            // Calling CreateDataFrame with schema
+            {
+                var data = new List<GenericRow>();
+                data.Add(new GenericRow(new object[] { "Alice", 20 }));
+                data.Add(new GenericRow(new object[] { "Bob", 30 }));
+
+                var schema = new StructType(new List<StructField>()
+                {
+                    new StructField("Name", new StringType()),
+                    new StructField("Age", new IntegerType())
+                });
+                DataFrame df = _spark.CreateDataFrame(data, schema);                
+                ValidateDataFrame<object[]>(data.ConvertAll(a => a.Values), df, schema);
+            }
+
+            // Calling CreateDataFrame(IEnumerable<string> _) without schema
+            {
+                var data = new List<string>(new string[] { "Alice", "Bob" });
+                var schema = new StructType(new List<StructField>()
+                {
+                    new StructField("_1", new StringType())
+                });
+
+                DataFrame df = _spark.CreateDataFrame(data);                
+                ValidateDataFrame<object[]>(data.ConvertAll(a => new object[] { a }), df, schema);
+            }
+
+            // Calling CreateDataFrame(IEnumerable<int> _) without schema
+            {
+                var data = new List<int>(new int[] { 1, 2 });
+                var schema = new StructType(new List<StructField>()
+                {
+                    new StructField("_1", new IntegerType())
+                });
+
+                DataFrame df = _spark.CreateDataFrame(data);
+                ValidateDataFrame<object[]>(data.ConvertAll(a => new object[] { a }), df, schema);
+            }
+
+            // Calling CreateDataFrame(IEnumerable<double> _) without schema
+            {
+                var data = new List<double>(new double[] { 1.2, 2.3 });
+                var schema = new StructType(new List<StructField>()
+                {
+                    new StructField("_1", new DoubleType())
+                });
+
+                DataFrame df = _spark.CreateDataFrame(data);
+                ValidateDataFrame<object[]>(data.ConvertAll(a => new object[] { a }), df, schema);
+            }
+
+            // Calling CreateDataFrame(IEnumerable<bool> _) without schema
+            {
+                var data = new List<bool>(new bool[] { true, false });
+                var schema = new StructType(new List<StructField>()
+                {
+                    new StructField("_1", new BooleanType())
+                });
+
+                DataFrame df = _spark.CreateDataFrame(data);
+                ValidateDataFrame<object[]>(data.ConvertAll(a => new object[] { a }), df, schema);
+            }
+        }
+
```

Comment:
`LongType()` missing.

Reply:
The dataframe for Long returns `int` instead of `long`. @imback82  Is this a known issue? It is working for all other types except Long.

Reply:
prob. related to #27. we can skip it for now.